### DISCUSSION TITLE
Starting a Keptn Governance Board

### DISCUSSION THREAD

**Discussion by oleg-nenashev** [keptn/keptn#8991] [Ideas]
Hi all. Since we're now an incubating project in the CNCF, we are expected to have a Governing Board and/or the technical steering committee. So far we have only the advisory board, and it is a different entity. I would suggest that we follow the requirements and start the entity.

### Suggested structure

- We start with a governing board. Any active contributor to Keptn is welcome to join. The definition for contributions is wide and includes any type of activity beneficial to the community: code contributions, documentation, active participation in channels, advocacy, and providing substantial feedback like GitHub Issues
- The board consists of 5 members, elected by an anonymous vote of community members. A sign-up form will be required
- None of the companies/organizations can have 50% seats or above. So there might be only 23 seats occupied by a single company
- Governance board operates asynchronously
- All Governance board members are invited to the quarterly Keptn Advisory Board meetings. They also get the status of Maintainers in the project and the respective voting rights in the CNCF

To make this approach fly in the current community state, we will need at least 3 active contributors from outside Dynatrace. It is doable but requires feedback and initial interest from usual suspects

 ### Responsibilities of a steering committee member could include

- Driving the project roadmap and facilitating key decisions
- Own the overall charter, vision, and direction of the project
- Promote broad adoption of the overall project
- Provide guidance for the project maintainers
- Actively participate in the steering committee meetings

Unlikely/nominal items:

- Manage membership of Keptn organization. We have de-facto community managers for it, e.g. me
- Archiving integrations or repositories of the Keptn organization and the ecosystem repositories. Same
- Approving and prioritizing new Keptn integrations

### References

- OpenFeature Governance model I suggest to follow: https://github.com/open-feature/community/blob/main/governance-charter.md
- Old tracking issue from @jetzlstorfer: https://github.com/keptn/community/issues/113

**Comment by oleg-nenashev** [comment]
If you're potentially interested in potentially joining the board, please comment here

**Reply by thschue** [reply]
I would be interested.

**Reply by vadasambar** [reply]
I would be interested

**Reply by bradmccoydev** [reply]
I would be interested.

**Reply by AloisReitbauer** [reply]
I would be more than interested :-)

**Reply by agardnerIT** [reply]
I would be interested

**Reply by braghettos** [reply]
I would be interested!

**Comment by thschue** [comment]
I generally like and support the initiative.

Nevertheless, I think the Governance Board members should have a fixed term (~1-2 years, but can be re-elected), and we should have some minimum qualifications, similar to crossplane -> https://github.com/crossplane/crossplane/blob/master/GOVERNANCE.md#steering-committee. Furthermore, I would restrict the voting to the current Governance Board and current maintainers of keptn/keptn (almost the same as in the crossplane doc), whereby everyone who fulfills the minimum qualifications should be able to get nominated. 

There could also be 1-2 "member-elected seats" for which all of the community members might be able to vote.

**Reply by oleg-nenashev** [reply]
> Furthermore, I would restrict the voting to the current Governance Board and current maintainers of keptn/keptn (almost the same as in the crossplane doc), whereby everyone who fulfills the minimum qualifications should be able to get nominated.

I respectfully disagree about restricting the vote to just maintainers and board members. First of all, the bootstrap Advisory board would not be an elected entity in the beginning, because it's an entity formed by Dynatrace its friends, without asking the community or announcing the existence of the entity. It's perfectly fine for that early stage of the project, but not for decision making. For maintainers, as you said, all of them are Dynatrace affiliated except Juergen. It is not possible to do a legitimate voting process just with this set of voters

**Reply by AloisReitbauer** [reply]
I am also for enabling non maintainers. We can define requirements for significant involvement in the project. @oleg-nenashev how many non-dynatrace project members are there. There should be enough to get a well balanced board.

**Reply by oleg-nenashev** [reply]
Yes, that's the plan. I am talking to people privately, but we should be good for the 5- or 7- members board.

**Comment by oleg-nenashev** [comment]
For the governance model, I would suggest to use the OpenFeature one as a starting point. It's a lightweight model focused on the open governance, and it's aligned with the CNCF common practices. It's based on OTel, Jenkins, Python governance models https://github.com/open-feature/community/blob/main/governance-charter.md

**Comment by mowies** [comment]
Maybe it would be good to give prospective board members the option between different term lengths upon nomination.
e.g. you can nominate yourself for a 1, 2 or 3 year term.
This would give nominees more options to make the commitment in a way they are maybe more comfortable with. Also, it would have the effect that we have overlapping terms which probably ensures a better overall steering capability since we don't get to a point where the whole board staffed with new people because all the previous terms ended at the same time.

**Comment by AnaMMedina21** [comment]
I am definitely interested!

**Comment by jetzlstorfer** [comment]
I am interested as well!

**Comment by thschue** [comment]
There is a vote ongoing for the setup of the governance committee (https://github.com/keptn/community/issues/183) and another issue open for the nominations (https://github.com/keptn/community/issues/184). Therefore, this thread will be closed.
